packages/flutter_tools/lib/src/web/compile.dart

```diff
@@ -199,6 +199,21 @@ enum WebRendererMode implements CliEnum {
   static const WebRendererMode defaultForJs = WebRendererMode.canvaskit;
   static const WebRendererMode defaultForWasm = WebRendererMode.skwasm;
 
+  /// Returns whether the WebRendererMode is considered deprecated or not.
+  ///
+  /// Deprecated modes: auto, html.
+  bool get isDeprecated => switch(this) {
+        auto => true,
+        canvaskit => false,
+        html => true,
+        skwasm => false
+      };
+
+  /// Returns a consistent deprecation warning for the WebRendererMode.
+  String get deprecationWarning =>
+    'The HTML Renderer is being deprecated. Stop using "--web-renderer=$name".'
```

Comment:
nit: at this point it's safe to say that it _is_ deprecated rather than _being_ deprecated. In the next version it will be removed, not deprecated.

Reply:
Yep, let me change that.